packages/react-composites/src/composites/CallWithChatComposite/adapter/CallWithChatAdapter.ts

```diff
@@ -65,7 +66,7 @@ export interface CallWithChatAdapterManagement {
    *
    * @public
    */
-  joinCall(microphoneOn?: boolean): Call | undefined;
+  joinCall(microphoneOn?: boolean): void;
```

Comment:
this looks like a breaking change

Reply:
This is for CommonCallAdapter - which is the interface(abstraction) needed by composite, it requires less so it is backward compatible 

For CallAdapter, it is remaining the same, joinCall still return Call | undefined there

Reply:
Right, but technically anyone using the `CallWithChatAdapterManagement` interface directly will now hit a breaking change.
In actuality, ARB wants us to have flat interfaces anyway, perhaps (best in a separate PR if we do) we leave this interface as is, mark as deprecated and not used, then update the whole interface to be flat (internally we can still use common interfaces to ensure type correctness across different adapters)

Reply:
Although I thought there would be very few people directly use CallWithChatAdapterManagement, but I still need to say, you are right, let's not lie to semver number, this is a breaking change.

I tried flattening apis, but it turned out to be a big change for api, and flattening those apis is not the focus of this PR, so I prefer not to do it here in the PR.

The solution to make minimum change here is that I made some changes to current api - I had to add one more interface to isolate "joinCall" "startCall" functions, then once flattening happens, we just need to deprecate all of those together and we are good.

And good news here - we don't have to make api changes to CallWithChat anymore